aws/resource_aws_default_vpc.go

```diff
@@ -15,6 +15,11 @@ func resourceAwsDefaultVpc() *schema.Resource {
 	dvpc.Create = resourceAwsDefaultVpcCreate
 	dvpc.Delete = resourceAwsDefaultVpcDelete
 
+	// Can't "terraform import" a Default VPC; Use "terraform apply"
```

Comment:
See https://github.com/terraform-providers/terraform-provider-aws/pull/4462.
Do we want to block `terraform import` of default VPC?

Reply:
Why would we need to _block_ import? It should be a valid form of managing an existing default VPC as with any other Terraform resource - although we just won't be performing any validation that it actually is or is not the default VPC on import currently. I think we can (and should) add that validation in the `aws_default_vpc` read function. Without this extra logic though, I think adding a documentation note that the import validation is not currently present is fine. Since the functionality is already present, some people might be depending on it already in their environment.

Honestly, the "magic" we have some places in the provider to cross-reference schema attributes, functions, etc. often makes everything more complicated in the end as seen here. I would almost prefer we just fully redefine the `aws_default_vpc` functions and schema to appropriately match the physical resource even if most of the logic is identical to `aws_vpc`.

Reply:
Agreed.
I think if possible it would be great to somehow combine the `aws_vpc` and `aws_default_vpc` resources (and maybe deprecate `aws_default_vpc`) especially given that default VPCs can now be re-created if they are deleted- https://github.com/terraform-providers/terraform-provider-aws/pull/1400?

On reflection this PR seems to be changing a lot of code for what on the face of it should be a relatively small (though important) change.
I'm happy to close this PR and spend some time thinking about a more maintainable path forward - All ideas welcome.